lighthouse-core/audits/byte-efficiency/unminified-javascript.js

```diff
@@ -78,24 +78,23 @@ class UnminifiedJavaScript extends ByteEfficiencyAudit {
     /** @type {Array<LH.Audit.ByteEfficiencyItem>} */
     const items = [];
     const warnings = [];
-    for (const {requestId, src, content} of artifacts.ScriptElements) {
-      if (!content) continue;
+    for (const script of artifacts.Scripts) {
+      if (!script.content) continue;
 
-      const networkRecord = networkRecords.find(record => record.requestId === requestId);
-      const displayUrl = !src || !networkRecord ?
-        `inline: ${content.substr(0, 40)}...` :
-        networkRecord.url;
+      const networkRecord = networkRecords.find(record => record.url === script.url);
+      const displayUrl = script.name === artifacts.URL.finalUrl ?
+        `inline: ${script.content.substring(0, 40)}...` :
+        script.name;
```

Comment:
note: `Script.name` could just be set to this in the gatherer so all audits benefit.